File: solution/027.remove-element/remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    int t = 0;
    int i = 0;
    for(i = 0;i < numsSize;)
    {
        if(nums[i] == val)
        {
            t = nums[numsSize - 1];
            nums[numsSize - 1] = nums[i];
            nums[i] = t;
            numsSize--;
        }
        else
            i++;
            
    }
    return numsSize;
}

File: solution/027.remove-element/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
        let i = 0,
          j = nums.length - 1,
          repeat = 0;
        do {
          if (nums[i] === val) {
            let temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            j--;
            repeat++;
          } else {
            i++;
          }
        } while (i <= j);
        nums.length = nums.length - repeat;
      };